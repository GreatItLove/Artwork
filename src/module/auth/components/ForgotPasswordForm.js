import React, { Component } from 'react';
import { Field } from 'redux-form';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import InputField from '../../common/components/form/InputField';
import SubmitButton from '../containers/SubmitButton';
import { LOGIN_PATH } from '../../../config/routesPatch';

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  forgotPassword: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired
};

class ForgotPasswordForm extends Component {
  handleFormSubmit = data => {
    this.props.forgotPassword(data);
  };

  render() {
    const { handleSubmit, t } = this.props;
    return (
      <div className="auth-login-wrapper">
        <div className="auth-login-fields">
          <div className="py-2" />
          <h4>{t('passwordForget.formHeader')}</h4>
          <div className="py-3" />
          <form onSubmit={handleSubmit(this.handleFormSubmit)}>
            <span>{t('passwordForget.headerInfo')}</span>
            <Field
              id="email"
              name="email"
              component={InputField}
              floatingLabelText={t('passwordForget.email.field')}
              type="email"
            />
            <div className="py-2" />
            <div className="pt20">
              <SubmitButton label={t('passwordForget.submitBtn')} />
            </div>
            <div className="py-2" />
            <div className="pt20 links">
              <Link to={LOGIN_PATH}>{t('passwordForget.backToLogin')}</Link>
            </div>
          </form>
          <div className="py-2" />
        </div>
      </div>
    );
  }
}

ForgotPasswordForm.propTypes = propTypes;

export default ForgotPasswordForm;
