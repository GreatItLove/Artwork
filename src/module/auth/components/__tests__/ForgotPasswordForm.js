/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ForgotPasswordForm from '../../components/ForgotPasswordForm';

describe('ForgotPassword component', () => {
  const t = key => key;
  const handleSubmit = jest.fn();
  const forgotPassword = jest.fn();
  const wrapper = shallow(<ForgotPasswordForm {...{ t, forgotPassword, handleSubmit }} />);
  it('Shallow snapshot initial state', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
