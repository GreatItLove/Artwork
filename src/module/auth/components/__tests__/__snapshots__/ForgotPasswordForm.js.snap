// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword component Shallow snapshot initial state 1`] = `
<div
  className="auth-login-wrapper"
>
  <div
    className="auth-login-fields"
  >
    <div
      className="py-2"
    />
    <h4>
      passwordForget.formHeader
    </h4>
    <div
      className="py-3"
    />
    <form>
      <span>
        passwordForget.headerInfo
      </span>
      <Field
        component={[Function]}
        floatingLabelText="passwordForget.email.field"
        id="email"
        name="email"
        type="email"
      />
      <div
        className="py-2"
      />
      <div
        className="pt20"
      >
        <Connect(SubmitBtnLoader)
          label="passwordForget.submitBtn"
        />
      </div>
      <div
        className="py-2"
      />
      <div
        className="pt20 links"
      >
        <Link
          replace={false}
          to="/login"
        >
          passwordForget.backToLogin
        </Link>
      </div>
    </form>
    <div
      className="py-2"
    />
  </div>
</div>
`;
